src/test/java/org/zalando/baigan/AnnotationBeanRegistrationIT.java

```diff
@@ -0,0 +1,45 @@
+package org.zalando.baigan;
+
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.zalando.baigan.annotation.ConfigurationServiceScan;
+import org.zalando.baigan.fixture.SomeConfiguration;
+import org.zalando.baigan.service.ConfigurationRepository;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {AnnotationBeanRegistrationIT.TestContext.class})
+public class AnnotationBeanRegistrationIT {
```

Comment:
Instead of having integration tests targeted at specific use cases, would it make sense to have a test for end-to-end functionality (like a component test for services), e.g. with localstack mocking the S3 part? Especially if we assume that bigger refactorings are coming, this could prove useful.

Reply:
I agree, a true E2E test, which effectively can act as an example/documentation, is the best way to go in the long-term https://github.com/zalando-stups/baigan-config/issues/73